### PR TITLE
Plugin settings refactor

### PR DESCRIPTION
**Pull request by SchrodingersGat**
Replaces https://github.com/inventree/InvenTree/pull/2478
Closes https://github.com/inventree/InvenTree/issues/2306

<a href="https://gitpod.io/#https://github.com/inventree/InvenTree/pull/2492"></a>

### CODE REVIEW

File: InvenTree/common/models.py [hunk 1]
```diff
@@ -977,13 +976,6 @@ def save(self, *args, **kwargs):
             'validator': bool,
             'requires_restart': True,
         },
-        'ENABLE_PLUGINS_GLOBALSETTING': {
```
Comment (matmair) [inline InvenTree/common/models.py:980]: @SchrodingersGat do you want plugins to have the right to define settings without enabling it first?

You will need to change this registry to, it checks if the setting is enabled:
 
 ​        ​if​ ​settings​.​PLUGIN_TESTING​ ​or​ ​InvenTreeSetting​.​get_setting​(​'ENABLE_PLUGINS_GLOBALSETTING'​):

**Review by matmair — CHANGES_REQUESTED** [review]
Looks great in general, pls check comment before merging.

File: InvenTree/common/models.py (hunk 1 above)
Comment (SchrodingersGat) [reply InvenTree/common/models.py:980]: Now that the settings are quarantined in their own table, I don't think this is required - it is perhaps an extra impediment to configuration?

**Comment by SchrodingersGat** [comment]
@matmair if all tests pass here, I *think* it is ready to be tested on your end :)

**Comment by SchrodingersGat** [comment]
@matmair let me know if you can see any outstanding issues here :)

File: InvenTree/common/models.py (hunk 1 above)
Comment (matmair) [reply InvenTree/common/models.py:980]: How should plugins share configs like website-endpoints etc. then? My idea is that if a plugin already exists for an integration with another service than a new plugin can extend on the functions of the first plugin an reuse settings set for it.
Do you want another mixin for marking settings globally available?

Comment (SchrodingersGat) [reply InvenTree/common/models.py:980]: good point, I had not considered that... Although there's nothing stopping a plugin for looking at settings provided by *other* plugins (just be key lookup)...

Comment (matmair) [reply InvenTree/common/models.py:980]: I think we will see in practice if that usecase pops up often enough to warrant a change / mixin.
I will rewrite my already existing plugins to use the new APIs and see if problems / code duplications arise.

Comment (SchrodingersGat) [reply InvenTree/common/models.py:980]: Fair enough, let's see how it plays out and refactor as necessary.

**Pull request merged**